crates/connectors/ndc-postgres/src/error/convert.rs

```diff
@@ -10,19 +10,22 @@ pub fn execution_error_to_query_error(
     match error {
         Error::Query(query_error) => match &query_error {
             QueryError::VariableNotFound(_) => {
-                connector::QueryError::InvalidRequest(query_error.to_string())
+                connector::QueryError::new_invalid_request(&"Invalid request")
+                    .with_details(serde_json::Value::String(query_error.to_string()))
             }
             QueryError::NotSupported(_) => {
-                connector::QueryError::UnsupportedOperation(query_error.to_string())
+                connector::QueryError::new_unsupported_operation(&"Unsupported operation")
+                    .with_details(serde_json::Value::String(query_error.to_string()))
             }
             QueryError::DBError(_) | QueryError::DBConstraintError(_) => {
-                connector::QueryError::UnprocessableContent(query_error.to_string())
-            }
-            QueryError::MutationConstraintFailed => {
-                connector::QueryError::Other(Box::new(query_error))
+                connector::QueryError::new_unprocessable_content(&"Unprocessable content")
+                    .with_details(serde_json::Value::String(query_error.to_string()))
             }
+            QueryError::MutationConstraintFailed => connector::QueryError::new("Internal error")
+                .with_details(serde_json::Value::String(query_error.to_string())),
         },
-        Error::DB(_) => connector::QueryError::Other(Box::new(error)),
+        Error::DB(_) => connector::QueryError::new("Internal error")
+            .with_details(serde_json::Value::String(error.to_string())),
```

Comment:
Details should be an object! you probably just want to pass in the error to the `new` function, and omit adding details

Reply:
I'd also be fine with this, avoiding the string message entirely and just getting to the error itself.